projects/Inventaire/module.nix

```diff
@@ -0,0 +1,157 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.inventaire;
+  settingsFormat = pkgs.formats.json { };
+  settingsJson = settingsFormat.generate "local.json" cfg.settings;
+  settingsCJS = pkgs.runCommandNoCC "local.cjs" { } ''
+    cat <<EOF >$out
+    // Settings generated via Nix
+
+    module.exports =
+    EOF
+    cat ${settingsJson} >> $out
+  '';
+
+  settingsLinks = ''
+    Links for settings (make sure to look at the version that's actually packaged):
+    - <a href="https://codeberg.org/inventaire/inventaire/src/branch/main/server/types/config.ts">Types</a>
+    - <a href="https://codeberg.org/inventaire/inventaire/src/branch/main/config/default.cjs">Defaults</a>
+  '';
+  mkExtraSettingsFileOption =
+    mode:
+    lib.options.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      description =
+        ''
+          Path to a .cjs file that exists at runtime with any additional setting.
+
+          Notes:
+          - The passed path (if any) will be set up as `config/local-${mode}.cjs`, to override the built-in defaults and values from the main `settings` attribute
+          - These settings are not world-readable (assuming you're passing a path that isn't generated by Nix), and thus suitable for secrets
+
+        ''
+        + settingsLinks;
+      default = null;
+    };
+in
+{
+  options.services.inventaire = {
+    enable = lib.options.mkEnableOption "Inventaire server";
+
+    package = lib.options.mkPackageOption pkgs "inventaire" { };
+
+    settings = lib.options.mkOption {
+      type = settingsFormat.type;
+      description =
+        ''
+          Settings that you wish to set for the service.
+
+          Notes:
+          - The settings here will be set up as `config/local.cjs`, to override the built-in defaults
+          - These settings are world-readable, and thus not suitable for secrets
+
+        ''
+        + settingsLinks;
+      default = {
+        hostname = "localhost";
+        port = 3006;
```

Comment:
This is awesome! Can we also have an `openFirewall` option as well? This is necessary for forwarding the port from the VM to the host.